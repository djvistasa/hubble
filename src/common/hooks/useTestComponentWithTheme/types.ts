type IUseTestComponentWithTheme = {};

export type { IUseTestComponentWithTheme };
