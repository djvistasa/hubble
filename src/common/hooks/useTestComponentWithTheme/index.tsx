import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";

import { RenderResult } from "@testing-library/react";

import { ReactElement } from "react";
import { QueryClient, QueryClientProvider } from "react-query";

function useTestComponentWithTheme() {
  const renderWithTheme = (component: ReactElement<any, any>): RenderResult => {
    const queryClient = new QueryClient();
    return render(
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </QueryClientProvider>
    );
  };

  return renderWithTheme;
}

export default useTestComponentWithTheme;
