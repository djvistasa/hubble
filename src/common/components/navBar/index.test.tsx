import useTestComponentWithTheme from "@hooks/useTestComponentWithTheme";
import { describe, expect, it } from "vitest";

import NavBar from ".";

describe("NavBar", () => {
  const mountWithTheme = useTestComponentWithTheme();

  it("should render correctly", () => {
    const component = mountWithTheme(<NavBar />);
    expect(component).toMatchSnapshot();
  });
});
