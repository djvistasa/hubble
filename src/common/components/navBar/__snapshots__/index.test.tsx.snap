// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__StyledNavBar-sc-1nmlvg1-0 cZtWZS"
      >
        <div
          class="styles__StyledNavBarInner-sc-1nmlvg1-1 fyjiyE"
        >
          <h1
            class="styles__StyledAppName-sc-1nmlvg1-3 gnDjXV"
          >
            Hubble Movies
          </h1>
          <div
            class="styles__StyledSearchBarWrapper-sc-1nmlvg1-2 eFQYtg"
          >
            <div
              class="styles__StyledInputWrapper-sc-27trcx-1 ePjCos"
            >
              <div
                class="styles__StyledInputInner-sc-27trcx-3 chZIvK"
              >
                <svg
                  fill="none"
                  height="20"
                  viewBox="0 0 100 100"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M39.5 73.9C19.9 73.9 4 58 4 38.5C4 19 19.9 3 39.5 3C59.1 3 75 18.9 75 38.5C75 58.1 59 73.9 39.5 73.9ZM39.5 20C29.3 20 21 28.3 21 38.5C21 48.7 29.3 57 39.5 57C49.7 57 58 48.7 58 38.5C58 28.3 49.7 20 39.5 20Z"
                    fill="#989099"
                  />
                  <path
                    d="M93.6998 82.9999L76.2999 65.4999C74.8999 64.0999 72.5999 64.1999 71.2999 65.6999C69.1999 68.0999 66.8999 70.2999 64.3999 72.1999C62.6999 73.3999 62.5999 75.8999 63.9999 77.2999L81.8999 95.1999C83.5999 96.8999 85.6999 97.6999 87.8999 97.6999C89.9999 97.6999 92.1999 96.8999 93.7999 95.2999C97.2999 91.9999 97.0998 86.3999 93.6998 82.9999Z"
                    fill="#989099"
                  />
                </svg>
                <input
                  class="styles__StyledInput-sc-27trcx-0 ecwbIm"
                  data-testid="input-test-search"
                  placeholder="Find movies, tv shows, people..."
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__StyledNavBar-sc-1nmlvg1-0 cZtWZS"
    >
      <div
        class="styles__StyledNavBarInner-sc-1nmlvg1-1 fyjiyE"
      >
        <h1
          class="styles__StyledAppName-sc-1nmlvg1-3 gnDjXV"
        >
          Hubble Movies
        </h1>
        <div
          class="styles__StyledSearchBarWrapper-sc-1nmlvg1-2 eFQYtg"
        >
          <div
            class="styles__StyledInputWrapper-sc-27trcx-1 ePjCos"
          >
            <div
              class="styles__StyledInputInner-sc-27trcx-3 chZIvK"
            >
              <svg
                fill="none"
                height="20"
                viewBox="0 0 100 100"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M39.5 73.9C19.9 73.9 4 58 4 38.5C4 19 19.9 3 39.5 3C59.1 3 75 18.9 75 38.5C75 58.1 59 73.9 39.5 73.9ZM39.5 20C29.3 20 21 28.3 21 38.5C21 48.7 29.3 57 39.5 57C49.7 57 58 48.7 58 38.5C58 28.3 49.7 20 39.5 20Z"
                  fill="#989099"
                />
                <path
                  d="M93.6998 82.9999L76.2999 65.4999C74.8999 64.0999 72.5999 64.1999 71.2999 65.6999C69.1999 68.0999 66.8999 70.2999 64.3999 72.1999C62.6999 73.3999 62.5999 75.8999 63.9999 77.2999L81.8999 95.1999C83.5999 96.8999 85.6999 97.6999 87.8999 97.6999C89.9999 97.6999 92.1999 96.8999 93.7999 95.2999C97.2999 91.9999 97.0998 86.3999 93.6998 82.9999Z"
                  fill="#989099"
                />
              </svg>
              <input
                class="styles__StyledInput-sc-27trcx-0 ecwbIm"
                data-testid="input-test-search"
                placeholder="Find movies, tv shows, people..."
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
